pop-api/examples/fungibles/tests.rs

```diff
@@ -0,0 +1,433 @@
+			.encode();
+	// A `Transfer` event is emitted.
+	assert_eq!(last_contract_event(&session).unwrap(), expected.as_slice());
+
+	assert_eq!(session.sandbox().total_supply(&TOKEN_ID), AMOUNT);
+	assert_eq!(session.sandbox().balance_of(&TOKEN_ID, &ALICE), AMOUNT);
+	Ok(())
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn burn_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+
+	const AMOUNT: Balance = TOKEN_MIN_BALANCE * 4;
+	// Mint tokens.
+	assert_ok!(session.sandbox().mint_into(&TOKEN_ID, &ALICE, AMOUNT));
+
+	// No-op.
+	assert_ok!(burn(&mut session, ALICE, 0));
+	assert_eq!(last_contract_event(&session), None); // No event emitted.
+
+	// Failed with `InsufficientBalance`.
+	expect_call_reverted(
+		&mut session,
+		BURN,
+		vec![ALICE.to_string(), (AMOUNT * 2).to_string()],
+		PSP22Error::InsufficientBalance,
+	);
+
+	// Successfully burn tokens.
+	assert_ok!(burn(&mut session, ALICE, 1));
+	let expected =
+		Transfer { from: Some(account_id_from_slice(ALICE.as_ref())), to: None, value: 1 }.encode();
+	// A `Transfer` event is emitted.
+	assert_eq!(last_contract_event(&session).unwrap(), expected.as_slice());
+
+	assert_eq!(session.sandbox().total_supply(&TOKEN_ID), AMOUNT - 1);
+	assert_eq!(session.sandbox().balance_of(&TOKEN_ID, &ALICE), AMOUNT - 1);
+	Ok(())
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn transfer_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	let contract = deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+
+	const AMOUNT: Balance = TOKEN_MIN_BALANCE * 4;
+	const TRANSFERRED: Balance = TOKEN_MIN_BALANCE;
+	// Mint tokens.
+	assert_ok!(session.sandbox().mint_into(&TOKEN_ID, &contract.clone(), AMOUNT));
+	assert_ok!(session.sandbox().mint_into(&TOKEN_ID, &BOB, AMOUNT));
+	// Check balance of accounts.
+	assert_eq!(session.sandbox().balance_of(&TOKEN_ID, &contract.clone()), AMOUNT);
+	assert_eq!(session.sandbox().balance_of(&TOKEN_ID, &BOB), AMOUNT);
+
+	session.set_actor(contract.clone());
+	// No-op if `value` is zero.
+	assert_ok!(transfer(&mut session, ALICE, 0));
+	assert_eq!(last_contract_event(&session), None); // No event emitted.
+
+	// No-op if the caller and `to` is the same address, returns success and no events are emitted.
+	assert_ok!(transfer(&mut session, contract.clone(), TRANSFERRED));
+	assert_eq!(last_contract_event(&session), None); // No event emitted.
+
+	// Failed with `InsufficientBalance`.
+	let data = serde_json::to_string::<[u8; 0]>(&[]).unwrap();
+	expect_call_reverted(
+		&mut session,
+		TRANSFER,
+		vec![BOB.to_string(), (AMOUNT + 1).to_string(), data],
+		PSP22Error::InsufficientBalance,
+	);
```

Comment:
This should be `assert_eq!(transfer(...), Err(PSP22Error::InsufficientBalance))`

Reply:
Same for all the other messages and tests, see `pop-api/integration-tests`